### PR TITLE
Ray activation fixes 4

### PR DESCRIPTION
**Pull request by mingwandroid**
Rebased and some of test_create.py ran OK on macOS, not tested yet beyond that.

### CODE REVIEW

File: tests/conda_env/test_cli.py [hunk 1]
```diff
@@ -77,26 +75,23 @@ def run_env_command(command, prefix, *arguments):
...
     elif command is Commands.ENV_CREATE: # CREATE
-        if prefix :
-            command_line = "{0} -f {1} {2}".format(command, prefix, " ".join(arguments))
-        else:
-            command_line = "{0} {1}".format(command, " ".join(arguments))
+        if prefix:
+            arguments[1:1] = ['-f', prefix]
     elif command is Commands.ENV_REMOVE:  # REMOVE
-        command_line = "{0} --yes -n {1} {2}".format(command, prefix, " ".join(arguments))
+        arguments[1:1] = ['--yes', '-n', prefix]
```
Comment (msarahan) [inline tests/conda_env/test_cli.py]: Is this a way to insert?  If so, could you please change to the .insert syntax?  This syntax is confusing to me.

File: conda/core/initialize.py [hunk 2]
```diff
@@ -621,9 +627,9 @@ def run_plan_elevated(plan):
             from ..common.os.windows import run_as_admin
             temp_path = None
             try:
-                with NamedTemporaryFile('w+b', suffix='.json', delete=False) as tf:
+                with Utf8NamedTemporaryFile('w+', suffix='.json', delete=False) as tf:
                     # the default mode is 'w+b', and universal new lines don't work in that mode
-                    tf.write(ensure_binary(json.dumps(plan, ensure_ascii=False)))
+                    tf.write(plan)
```
Comment (nehaljwani) [inline conda/core/initialize.py]: Here, `type(plan)` is `list`, so if by some chance this code path is hit, there is a traceback.
PY3:
```
(base) C:\Users\Wani\repos\aggregate>conda init

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "C:\Users\Wani\repos\conda\conda\exceptions.py", line 1003, in __call__
        return func(*args, **kwargs)
      File "C:\Users\Wani\repos\conda\conda\cli\main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "C:\Users\Wani\repos\conda\conda\cli\conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "C:\Users\Wani\repos\conda\conda\cli\main_init.py", line 52, in execute
        anaconda_prompt, args.reverse)
      File "C:\Users\Wani\repos\conda\conda\core\initialize.py", line 111, in initialize
        run_plan_elevated(plan2)
      File "C:\Users\Wani\repos\conda\conda\core\initialize.py", line 632, in run_plan_elevated
        tf.write(plan)
      File "C:\Users\Wani\scratch\dev3\lib\tempfile.py", line 481, in func_wrapper
        return func(*args, **kwargs)
    TypeError: write() argument must be str, not list
```

PY2:
```
(C:\Users\Wani\scratch\dev) C:\Users\Wani\repos\conda>conda init
No handlers could be found for logger "conda._vendor.auxlib.packaging"
WARNING: The conda.compat module is deprecated and will be removed in a future release.

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "c:\users\wani\repos\conda\conda\exceptions.py", line 1003, in __call__
        return func(*args, **kwargs)
      File "c:\users\wani\repos\conda\conda\cli\main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "c:\users\wani\repos\conda\conda\cli\conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "c:\users\wani\repos\conda\conda\cli\main_init.py", line 52, in execute
        anaconda_prompt, args.reverse)
      File "c:\users\wani\repos\conda\conda\core\initialize.py", line 105, in initialize
        run_plan_elevated(plan1)
      File "c:\users\wani\repos\conda\conda\core\initialize.py", line 628, in run_plan_elevated
        tf.write(plan)
      File "C:\Users\Wani\scratch\dev\lib\codecs.py", line 369, in write
        data, consumed = self.encode(object, self.errors)
    TypeError: coercing to Unicode: need string or buffer, list found
```

However, if we make it dump json, then it fails to serialize one of the arguments to one of the functions:
```
{  
   'function':'init_cmd_exe_registry',
   'kwargs':{  
      'target_path':'HKEY_CURRENT_USER\\Software\\Microsoft\\Command Processor\\AutoRun',
      'conda_prefix':'C:\\Users\\Wani\\scratch\\dev3',
      'reverse':<auxlib._Null object at 0x0000019F60A0E080> <---------- 😭 
   },
   'result':'no change'
}
```

Comment (msarahan) [reply conda/core/initialize.py]: I think the right answer is to fix the _Null to a JSON-serializable type, or perhaps teach the JSON serializer how to serialize that particular type, and to dump to JSON.

File: conda/_vendor/auxlib/__init__.py [hunk 3]
```diff
@@ -72,6 +72,11 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(_Null)
 
+    def __json__(self):
+        return 'null'
+
+    to_json = __json__
+
```
Comment (nehaljwani) [inline conda/_vendor/auxlib/__init__.py]: This is still not sufficient:
```
>>> from conda._vendor import auxlib
>>> import json
>>> a['a'].__json__()
'null'
>>> a
{'a': <auxlib._Null object at 0x109d10be0>}
>>> json.dumps(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/nwani/m3/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/Users/nwani/m3/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Users/nwani/m3/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Users/nwani/m3/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type _Null is not JSON serializable
```

Comment (nehaljwani) [reply conda/_vendor/auxlib/__init__.py]: https://bugs.python.org/issue27362

File: conda/core/initialize.py (hunk 2 above)
Comment (nehaljwani) [reply conda/core/initialize.py]: Okay, this diff seems to make it work:
```patch
diff --git a/conda/core/initialize.py b/conda/core/initialize.py
index 1fc6caa53..451e9651a 100644
--- a/conda/core/initialize.py
+++ b/conda/core/initialize.py
@@ -629,7 +629,8 @@ def run_plan_elevated(plan):
             try:
                 with Utf8NamedTemporaryFile('w+', suffix='.json', delete=False) as tf:
                     # the default mode is 'w+b', and universal new lines don't work in that mode
-                    tf.write(plan)
+                    tf.write(json.dumps(plan, ensure_ascii=False,
+                                        default=lambda x: x.__dict__))
                     temp_path = tf.name
                 python_exe = '"%s"' % abspath(sys.executable)
                 hinstance, error_code = run_as_admin((python_exe, '-m',  'conda.core.initialize',
```

**Comment by msarahan** [comment]
We are investigating the remaining windows failure, but believe it is a problem in our tests.  Merging.

**Pull request merged**